src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -13957,6 +14855,169 @@ public static Vector512<T> Invoke(Vector512<T> x, Vector512<T> y)
+            {
+                if (sizeof(T) == 1) return (Vector128.ShiftRightLogical(x.AsByte(), _amount & 7) | (Vector128.ShiftLeft(x.AsByte(), (8 - _amount) & 7))).As<byte, T>();
+                if (sizeof(T) == 2) return (Vector128.ShiftRightLogical(x.AsUInt16(), _amount & 15) | (Vector128.ShiftLeft(x.AsUInt16(), (16 - _amount) & 15))).As<ushort, T>();
+                if (sizeof(T) == 4) return (Vector128.ShiftRightLogical(x.AsUInt32(), _amount) | (Vector128.ShiftLeft(x.AsUInt32(), 32 - _amount))).As<uint, T>();
+                if (sizeof(T) == 8) return (Vector128.ShiftRightLogical(x.AsUInt64(), _amount) | (Vector128.ShiftLeft(x.AsUInt64(), 64 - _amount))).As<ulong, T>();
+                throw new NotSupportedException();
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public Vector256<T> Invoke(Vector256<T> x)
+            {
+                if (sizeof(T) == 1) return (Vector256.ShiftRightLogical(x.AsByte(), _amount & 7) | (Vector256.ShiftLeft(x.AsByte(), (8 - _amount) & 7))).As<byte, T>();
+                if (sizeof(T) == 2) return (Vector256.ShiftRightLogical(x.AsUInt16(), _amount & 15) | (Vector256.ShiftLeft(x.AsUInt16(), (16 - _amount) & 15))).As<ushort, T>();
+                if (sizeof(T) == 4) return (Vector256.ShiftRightLogical(x.AsUInt32(), _amount) | (Vector256.ShiftLeft(x.AsUInt32(), 32 - _amount))).As<uint, T>();
+                if (sizeof(T) == 8) return (Vector256.ShiftRightLogical(x.AsUInt64(), _amount) | (Vector256.ShiftLeft(x.AsUInt64(), 64 - _amount))).As<ulong, T>();
+                throw new NotSupportedException();
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public Vector512<T> Invoke(Vector512<T> x)
+            {
+                if (sizeof(T) == 1) return (Vector512.ShiftRightLogical(x.AsByte(), _amount & 7) | (Vector512.ShiftLeft(x.AsByte(), (8 - _amount) & 7))).As<byte, T>();
+                if (sizeof(T) == 2) return (Vector512.ShiftRightLogical(x.AsUInt16(), _amount & 15) | (Vector512.ShiftLeft(x.AsUInt16(), (16 - _amount) & 15))).As<ushort, T>();
+                if (sizeof(T) == 4) return (Vector512.ShiftRightLogical(x.AsUInt32(), _amount) | (Vector512.ShiftLeft(x.AsUInt32(), 32 - _amount))).As<uint, T>();
+                if (sizeof(T) == 8) return (Vector512.ShiftRightLogical(x.AsUInt64(), _amount) | (Vector512.ShiftLeft(x.AsUInt64(), 64 - _amount))).As<ulong, T>();
+                throw new NotSupportedException();
+            }
+        }
+
+        /// <summary>T.ScaleB(x, n)</summary>
+        internal readonly struct ScaleBOperator<T>(int n) : IStatefulUnaryOperator<T> where T : IFloatingPointIeee754<T>
+        {
+            private readonly int _n = n;
+
+            public static bool Vectorizable => false; // TODO: Vectorize
+
+            public T Invoke(T x) => T.ScaleB(x, _n);
+
+            public Vector128<T> Invoke(Vector128<T> x) => throw new NotSupportedException();
```

Comment:
`ScaleB` is mapping to the CRT `scalbn` which is just `x * 2^n`. If we want a cheap and mostly correct API, this would work until I can get the proper vectorization up (similar idea to what you're trying in https://github.com/dotnet/runtime/pull/97623).

Reply:
There may be some edges where IEEE 754 requires a particular behavior that might not be entirely handled correctly, but tests should identify those.

For example, `scalbn` expects `x == +/-0`, `x == +/-inf`, and `n == 0` to return `exp` unmodified